### ISSUE TITLE
fatal error: concurrent map read and map write

### ISSUE THREAD

**Issue by trumblejoe** [rclone/rclone#2378]
Hello, I just recieved this error. Details below.

What is your rclone version (eg output from rclone -V)
rclone v1.41 -os/arch linux/amd64 -go version go1.10

Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux lw701 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1+deb9u1 (2018-05-07) x86_64 GNU/Linux

Which cloud storage system are you using? (eg Google Drive)
Google Drive

The command you were trying to run (eg rclone copy /tmp remote:tmp)
Plex Scan of newly created library.

Here are my mount commands, rclone cache of encrypted remote files, with gocryptfs to decrypt the local cache per recommended best practices.

rclone mount gdriveCryptCache: files/rclone/gdriveCrypt 
--allow-other 
--dir-cache-time 160h 
--vfs-cache-mode=writes 
--cache-chunk-size 10M 
--cache-info-age 168h 
--cache-workers 8 
--buffer-size 0M 
--fuse-flag sync_read 
--umask 000 
--log-level INFO 
--log-file .tmp/gdriveCryptCache-cache.log &

gocryptfs -allow_other -ko sync "files/rclone/gdriveCrypt" "files/rclone/gdriveCryptPlain"

[gdriveCryptCache-cache[1].log](https://github.com/ncw/rclone/files/2164860/gdriveCryptCache-cache.1.log)

My logs are attached.

**Comment by ncw** [comment]
This report is for 1.41 but the code looks unchanged in master.  Can you get it to do this often or was it just a one-off?

The log + traceback looks like this

```
2018/07/04 21:08:45 ERROR : hZHyminAJ2S4WIDoc5Qlrg==/nnzGs0CkaF0QP1b-Z1DfWw==/Yg45xm4Jj-zl3YWMUI6qgw==/7cH5GbC_BcCmtNc2RBlZvaVL0dCDQIOzhjKZy7UGqXA=/4bg5QamlFkK16-kMFum4VJtfiUKazgl9AlM4M0tUYLkSS9Jao_r4i4Cb9zC3iAA0: unexpected conditions during reading. current position: 3388866560, current chunk position: 3386900480, current chunk size: 1449984, offset: 1966080, chunk size: 10485760, file size: 6881591176
2018/07/04 21:08:45 ERROR : hZHyminAJ2S4WIDoc5Qlrg==/nnzGs0CkaF0QP1b-Z1DfWw==/Yg45xm4Jj-zl3YWMUI6qgw==/7cH5GbC_BcCmtNc2RBlZvaVL0dCDQIOzhjKZy7UGqXA=/4bg5QamlFkK16-kMFum4VJtfiUKazgl9AlM4M0tUYLkSS9Jao_r4i4Cb9zC3iAA0: ReadFileHandle.Read error: low level retry 1/10: unexpected EOF

fatal error: concurrent map writes

goroutine 140156 [running]:
runtime.throw(0x103333c, 0x15)
	/opt/go/go1.10/src/runtime/panic.go:619 +0x81 fp=0xc4223e17e0 sp=0xc4223e17c0 pc=0x42b131
runtime.mapassign_fast64(0xea5d60, 0xc424a4f3e0, 0x1, 0x20)
	/opt/go/go1.10/src/runtime/hashmap_fast.go:531 +0x2f6 fp=0xc4223e1828 sp=0xc4223e17e0 pc=0x40b7e6
github.com/ncw/rclone/backend/cache.(*Object).Hash(0xc4206b2140, 0x1, 0xc4223e1970, 0xc4208ae418, 0xc4208ae420, 0x410e88)
	/home/ncw/go/src/github.com/ncw/rclone/backend/cache/object.go:318 +0x13a fp=0xc4223e18a8 sp=0xc4223e1828 pc=0x8da76a
github.com/ncw/rclone/vfs.(*ReadFileHandle).checkHash(0xc420302380, 0x0, 0x0)
	/home/ncw/go/src/github.com/ncw/rclone/vfs/read.go:274 +0x150 fp=0xc4223e19e0 sp=0xc4223e18a8 pc=0xcead70
github.com/ncw/rclone/vfs.(*ReadFileHandle).close(0xc420302380, 0xc420692780, 0x103f890)
	/home/ncw/go/src/github.com/ncw/rclone/vfs/read.go:338 +0xa6 fp=0xc4223e1a10 sp=0xc4223e19e0 pc=0xceb176
github.com/ncw/rclone/vfs.(*ReadFileHandle).Release(0xc420302380, 0x0, 0x0)
	/home/ncw/go/src/github.com/ncw/rclone/vfs/read.go:388 +0xea fp=0xc4223e1a80 sp=0xc4223e1a10 pc=0xceb51a
github.com/ncw/rclone/cmd/mount.(*FileHandle).Release(0xc4201a2520, 0x11381e0, 0xc4222f37c0, 0xc4221b6be0, 0xc4201a2520, 0x1001)
	/home/ncw/go/src/github.com/ncw/rclone/cmd/mount/handle.go:83 +0x11b fp=0xc4223e1ae0 sp=0xc4223e1a80 pc=0xcfde8b
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).handleRequest(0xc42071c0c0, 0x11381e0, 0xc4222f37c0, 0x112c840, 0xc42000c088, 0xc421388940, 0x1136720, 0xc4221b6be0, 0xc4223e1ef8, 0xc4220b9f80, ...)
	/home/ncw/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:1301 +0x2a08 fp=0xc4223e1e10 sp=0xc4223e1ae0 pc=0xcd9568
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve(0xc42071c0c0, 0x1136720, 0xc4221b6be0)
	/home/ncw/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:878 +0x41e fp=0xc4223e1fa0 sp=0xc4223e1e10 pc=0xcd673e
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve.func1(0xc42071c0c0, 0x1136720, 0xc4221b6be0)
	/home/ncw/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:425 +0x6e fp=0xc4223e1fc8 sp=0xc4223e1fa0 pc=0xcdb01e
runtime.goexit()
	/opt/go/go1.10/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc4223e1fd0 sp=0xc4223e1fc8 pc=0x458c41
created by github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve
	/home/ncw/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:423 +0x331
```

Which points at this code.  What I suspect is Hash() is being called concurrently and both calls are making a new Hash at the same time.

```go
// Hash requests a hash of the object and stores in the cache
// since it might or might not be called, this is lazy loaded
func (o *Object) Hash(ht hash.Type) (string, error) {
	_ = o.refresh()
	if o.CacheHashes == nil {
		o.CacheHashes = make(map[hash.Type]string)
	}

	cachedHash, found := o.CacheHashes[ht]
	if found {
		return cachedHash, nil
	}
	if err := o.refreshFromSource(false); err != nil {
		return "", err
	}
	liveHash, err := o.Object.Hash(ht)
	if err != nil {
		return "", err
	}
	o.CacheHashes[ht] = liveHash // <- PROBLEM

	o.persist()
	fs.Debugf(o, "object hash cached: %v", liveHash)

	return liveHash, nil
}
```

@remusb thoughts?

**Comment by trumblejoe** [comment]
I've had this happen twice in an eight-hour period. (I've been using rclone cache mount for about thirty-six  hours as of this morning). I've tried various settings but it did not start happening until I added --fuse-flag sync_read to the rclone command, and -ko sync to the gocryptfs command.

My suspicion is the -ko sync option being passed is mucking things up. Gocryptfs has a -serialize_reads option which may be a better suited but I am waiting on an admin to update my gocryptfs installation before I can try that with --fuse-flag sync_read.

After this error I dismounted everything and am now running rclone mount with --fuse-flag sync_read and gocryptfs without -ko sync, and the eight hours it ran overnight, it has not crashed yet.

The reason I've added those options is that it seems to greatly improve accessing the cache when it is a network storage on the other end. Though from what I've read top-end throughput speeds could be brought down.

I like what you mentioned about the hash firing concurrently. Another thing to consider is that gocryptfs is file-level encryption, and it has to read each .dirIV file, and the folder/contents; it's a lot going on.

**Comment by ncw** [comment]
The problem has a fairly obvious cause and fix so I don't think it will be difficult to fix since we have a traceback.

I'd like to hear what @remusb has to say about the problem though.

**Comment by systemmonkey42** [comment]
Hi, I started an rclone session on windows with a cache backed sftp remote, and almost immediately got this error

```
>rclone -v mount --read-only remote:/home/git j:
2019/06/04 19:26:19 INFO  : remote: Cache DB path: C:\Users\redacted\AppData\Local\rclone\cache-backend\remote.db
2019/06/04 19:26:19 INFO  : remote: Cache chunk path: C:\Users\redacted\AppData\Local\rclone\cache-backend\remote
2019/06/04 19:26:19 INFO  : remote: Chunk Memory: true
2019/06/04 19:26:19 INFO  : remote: Chunk Size: 5M
2019/06/04 19:26:19 INFO  : remote: Chunk Total Size: 10G
2019/06/04 19:26:19 INFO  : remote: Chunk Clean Interval: 1m0s
2019/06/04 19:26:19 INFO  : remote: Workers: 4
2019/06/04 19:26:19 INFO  : remote: File Age: 6h0m0s
2019/06/04 19:26:19 INFO  : Cache remote remote:home/git: poll-interval is not supported by this remote
The service rclone has been started.
fatal error: concurrent map writes

goroutine 232 [running, locked to thread]:
runtime.throw(0x133ceb5, 0x15)
        C:/go/src/runtime/panic.go:608 +0x79 fp=0xc000a21b70 sp=0xc000a21b40 pc=0x430799
runtime.mapassign_fast64(0x1140a40, 0xc000111ce0, 0x1, 0x20)
        C:/go/src/runtime/map_fast64.go:101 +0x30c fp=0xc000a21bb0 sp=0xc000a21b70 pc=0x41320c
github.com/ncw/rclone/backend/cache.(*Object).Hash(0xc0007c1c20, 0x1, 0xc000a21ce8, 0x0, 0xc000060040, 0x5460)
        C:/gopath/src/github.com/ncw/rclone/backend/cache/object.go:326 +0x12a fp=0xc000a21c28 sp=0xc000a21bb0 pc=0x98c9ea
github.com/ncw/rclone/vfs.(*ReadFileHandle).checkHash(0xc0004b42a0, 0x13a25f0, 0xc0004b42a0)
        C:/gopath/src/github.com/ncw/rclone/vfs/read.go:292 +0x153 fp=0xc000a21d58 sp=0xc000a21c28 pc=0xeac183
github.com/ncw/rclone/vfs.(*ReadFileHandle).Flush(0xc0004b42a0, 0x0, 0x0)
        C:/gopath/src/github.com/ncw/rclone/vfs/read.go:382 +0x85 fp=0xc000a21dc0 sp=0xc000a21d58 pc=0xeac745
github.com/ncw/rclone/cmd/cmount.(*FS).Flush(0xc000136d00, 0xc00095cce0, 0x10, 0x81, 0xc0006203c0)
        C:/gopath/src/github.com/ncw/rclone/cmd/cmount/fs.go:380 +0x1f1 fp=0xc000a21e40 sp=0xc000a21dc0 pc=0xec1801
github.com/ncw/rclone/vendor/github.com/billziss-gh/cgofuse/fuse.hostFlush(0x2240f10, 0x690fe40, 0x0)
        C:/gopath/src/github.com/ncw/rclone/vendor/github.com/billziss-gh/cgofuse/fuse/host.go:829 +0xb2 fp=0xc000a21e88 sp=0xc000a21e40 pc=0xe99212
github.com/ncw/rclone/vendor/github.com/billziss-gh/cgofuse/fuse._cgoexpwrap_b0480560eb0d_hostFlush(0x2240f10, 0x690fe40, 0x80)
        _cgo_gotypes.go:749 +0x3c fp=0xc000a21eb0 sp=0xc000a21e88 pc=0xe966dc
runtime.call32(0x0, 0x690fc70, 0x690fde0, 0x18)
        C:/go/src/runtime/asm_amd64.s:522 +0x42 fp=0xc000a21ee0 sp=0xc000a21eb0 pc=0x45cb42
runtime.cgocallbackg1(0x0)
        C:/go/src/runtime/cgocall.go:316 +0x1a9 fp=0xc000a21f58 sp=0xc000a21ee0 pc=0x404739
runtime.cgocallbackg(0x0)
        C:/go/src/runtime/cgocall.go:194 +0xeb fp=0xc000a21fc0 sp=0xc000a21f58 pc=0x4044eb
runtime.cgocallback_gofunc(0x0, 0x0, 0x0, 0x0)
        C:/go/src/runtime/asm_amd64.s:775 +0xb2 fp=0xc000a21fe0 sp=0xc000a21fc0 pc=0x45e1a2
runtime.goexit()
        C:/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000a21fe8 sp=0xc000a21fe0 pc=0x45e931

```

This is my entire config file.  The remote host is over a slowish internet connection which is why caching seemed a good idea.

```
[_remote]
type = sftp
host = remote
user = root
key_use_agent = true

[remote]
type = cache
remote = _remote:

```

Am i doing something obviously wrong?

Thanks

**Comment by ncw** [comment]
> Am i doing something obviously wrong?

No it is a bug.

Can you try this fix?

https://beta.rclone.org/branch/v1.47.0-097-gf46c5bc1-fix-2378-cache-crash-beta/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
Is this bug still relevant?

Did you have a go with the beta posted above?

Thanks

**Comment by jpkrohling** [comment]
@eclipseo, @ncw, how hard would it be to get this fix packaged as an update for Fedora 31? Looks like Fedora 31 is at 1.50 ([`rclone-1.50.2-1.fc31`](https://koji.fedoraproject.org/koji/buildinfo?buildID=1428911)), and this fix should be available as part of 1.51.

**Comment by ncw** [comment]
@jpkrohling I don't have any contact with the Fedora maintainer alas. The patch for this is https://github.com/rclone/rclone/commit/08a395788091a7c1a3b277115b7ff79563da8cfd

If you want you can use the 1.51 rpm package from here: https://rclone.org/downloads/

**Comment by jpkrohling** [comment]
Thanks, I wasn't aware of those packages. I'll try it out.

**Comment by jpkrohling** [comment]
Interesting: looks like I'm either hitting a new bug, or this bug here has appeared somewhere else:

```console
$ rclone copy 07-Upload/ photos-kroehling:/upload/  -P
Transferred:        5.377G / 30.191 GBytes, 18%, 1.045 MBytes/s, ETA 6h45m6s
Transferred:         2224 / 12231, 18%
Elapsed time:   1h27m47.4s
Transferring:
 *             200810/IMG_0637-52654f45568d2.jpg:100% /3.359M, 343.897k/s, 0s
 *             200810/IMG_0638-52654fe8e84a4.jpg:100% /3.436M, 351.827k/s, 0s
 *             200810/IMG_0639-52654ee0391d5.jpg: 87% /3.207M, 714.993k/s, 0s
 *             200810/IMG_0640-52654ff016688.jpg: 22% /3.150M, 0/s, -fatal error: concurrent map writes

goroutine 90 [running]:
runtime.throw(0x16d69cf, 0x15)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/panic.go:774 +0x72 fp=0xc0006516a0 sp=0xc000651670 pc=0x42f522
runtime.mapassign_faststr(0x15f0aa0, 0xc00034b590, 0xc000c579b0, 0x6, 0x40a4ea)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/map_faststr.go:211 +0x417 fp=0xc000651708 sp=0xc0006516a0 pc=0x4135b7
github.com/rclone/rclone/fs/dirtree.DirTree.Add(0xc00034b590, 0x1a28500, 0xc00119a2a0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/dirtree/dirtree.go:37 +0x92 fp=0xc000651760 sp=0xc000651708 pc=0x941842
github.com/rclone/rclone/fs/dirtree.DirTree.AddEntry(0xc00034b590, 0x1a28500, 0xc00119a2a0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/dirtree/dirtree.go:59 +0x137 fp=0xc0006517a0 sp=0xc000651760 pc=0x941b67
github.com/rclone/rclone/backend/googlephotos.(*Object).Update(0xc00119a2a0, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc00085e8f0, 0x7f27628b38b8, 0xc000c7d380, 0xc000092e60, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:1006 +0x9bb fp=0xc000651b38 sp=0xc0006517a0 pc=0xd23c2b
github.com/rclone/rclone/backend/googlephotos.(*Fs).Put(0xc0005b7b90, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc00085e8f0, 0x7f27628b38b8, 0xc000c7d380, 0xc000092e60, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:623 +0x218 fp=0xc000651c28 sp=0xc000651b38 pc=0xd20218
github.com/rclone/rclone/fs/operations.Copy(0x1a28100, 0xc0005a7ec0, 0x1a3d120, 0xc0005b7b90, 0x0, 0x0, 0xc000c579b0, 0x21, 0x1a3d620, 0xc000c7d380, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:425 +0x152c fp=0xc000651ed0 sp=0xc000651c28 pc=0x9aa20c
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairCopyOrMove(0xc0002b61c0, 0x1a28100, 0xc0005a7ec0, 0xc0005a8dc0, 0x1a3d120, 0xc0005b7b90, 0xc0002b6298)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:331 +0x21c fp=0xc000651fa8 sp=0xc000651ed0 pc=0x118144c
runtime.goexit()
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc000651fb0 sp=0xc000651fa8 pc=0x45f131
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startTransfers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:356 +0xb1

goroutine 1 [semacquire, 87 minutes]:
sync.runtime_Semacquire(0xc000132b28)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/sema.go:56 +0x42
sync.(*WaitGroup).Wait(0xc000132b20)
	/opt/hostedtoolcache/go/1.13.7/x64/src/sync/waitgroup.go:130 +0x64
github.com/rclone/rclone/fs/march.(*March).Run(0xc0005b7c20, 0xc0005b7c20, 0x1a32ea0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:201 +0x31b
github.com/rclone/rclone/fs/sync.(*syncCopyMove).run(0xc0002b61c0, 0xc0000400b8, 0x1a3d120)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:706 +0x20b
github.com/rclone/rclone/fs/sync.runSyncCopyMove(0x1a28140, 0xc0000400b8, 0x1a3d120, 0xc0005b7b90, 0x1a3d5a0, 0xc00010d700, 0xc000000000, 0xc000132ac0, 0xc00004ed20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:927 +0x276
github.com/rclone/rclone/fs/sync.CopyDir(...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:937
github.com/rclone/rclone/cmd/copy.glob..func1.1(0xc0001557a0, 0x1753fe8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/cmd/copy/copy.go:81 +0x85
github.com/rclone/rclone/cmd.Run(0xc000340101, 0x24ad7e0, 0xc00133bd10)
	/home/runner/work/rclone/src/github.com/rclone/rclone/cmd/cmd.go:241 +0xc6
github.com/rclone/rclone/cmd/copy.glob..func1(0x24ad7e0, 0xc00034adb0, 0x2, 0x3)
	/home/runner/work/rclone/src/github.com/rclone/rclone/cmd/copy/copy.go:79 +0x129
github.com/rclone/rclone/vendor/github.com/spf13/cobra.(*Command).execute(0x24ad7e0, 0xc00034acf0, 0x3, 0x3, 0x24ad7e0, 0xc00034acf0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/github.com/spf13/cobra/command.go:830 +0x2aa
github.com/rclone/rclone/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x24aa0e0, 0x5e54d359, 0x24c19a0, 0xc00004e178)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/rclone/rclone/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/github.com/spf13/cobra/command.go:864
github.com/rclone/rclone/cmd.Main()
	/home/runner/work/rclone/src/github.com/rclone/rclone/cmd/cmd.go:506 +0x92
main.main()
	/home/runner/work/rclone/src/github.com/rclone/rclone/rclone.go:14 +0x20

goroutine 6 [syscall, 87 minutes]:
os/signal.signal_recv(0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/sigqueue.go:147 +0x9c
os/signal.loop()
	/opt/hostedtoolcache/go/1.13.7/x64/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/opt/hostedtoolcache/go/1.13.7/x64/src/os/signal/signal_unix.go:29 +0x41

goroutine 18 [select]:
github.com/rclone/rclone/vendor/go.opencensus.io/stats/view.(*worker).start(0xc000162960)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/go.opencensus.io/stats/view/worker.go:154 +0x100
created by github.com/rclone/rclone/vendor/go.opencensus.io/stats/view.init.0
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/go.opencensus.io/stats/view/worker.go:32 +0x57

goroutine 14 [select]:
github.com/rclone/rclone/cmd.startProgress.func2(0xc000132ac0, 0xc00004ed20, 0x1751a28)
	/home/runner/work/rclone/src/github.com/rclone/rclone/cmd/progress.go:48 +0x132
created by github.com/rclone/rclone/cmd.startProgress
	/home/runner/work/rclone/src/github.com/rclone/rclone/cmd/progress.go:40 +0xcc

goroutine 15 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 16 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 81 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 82 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 83 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 84 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 85 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 86 [select, 87 minutes]:
github.com/rclone/rclone/fs/sync.(*pipe).Get(0xc0005a8d70, 0x1a28100, 0xc0005a7ec0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:117 +0xe4
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc0002b61c0, 0xc0005a8d70, 0xc0005a8dc0, 0xc0002b6280)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:247 +0xb8
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:341 +0x9a

goroutine 87 [select]:
net/http.(*http2ClientConn).roundTrip(0xc001051e00, 0xc00026cd00, 0x0, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:7573 +0x9a0
net/http.(*http2Transport).RoundTripOpt(0xc0005b9320, 0xc00026cd00, 0xc00077aa00, 0x725fc5, 0x0, 0xc00046a360)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6936 +0x172
net/http.(*http2Transport).RoundTrip(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6897
net/http.http2noDialH2RoundTripper.RoundTrip(0xc0005b9320, 0xc00026cd00, 0xc000b7c270, 0x5, 0xc00046a3e8)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:9032 +0x3e
net/http.(*Transport).roundTrip(0xc00016c280, 0xc00026cd00, 0x16c4b8f, 0xa, 0xc000827db0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/transport.go:485 +0xdc1
net/http.(*Transport).RoundTrip(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/roundtrip.go:17
github.com/rclone/rclone/fs/fshttp.(*Transport).RoundTrip(0xc00034b380, 0xc00026cd00, 0xc00026cd00, 0x0, 0x150)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:350 +0x21a
github.com/rclone/rclone/vendor/golang.org/x/oauth2.(*Transport).RoundTrip(0xc00034b4a0, 0xc00026cc00, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/golang.org/x/oauth2/transport.go:56 +0x124
net/http.send(0xc00026cc00, 0x19f7cc0, 0xc00034b4a0, 0x0, 0x0, 0x0, 0xc001276248, 0x203000, 0x1, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:250 +0x443
net/http.(*Client).send(0xc00034b4d0, 0xc00026cc00, 0x0, 0x0, 0x0, 0xc001276248, 0x0, 0x1, 0x1693780)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:174 +0xfa
net/http.(*Client).do(0xc00034b4d0, 0xc00026cc00, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:641 +0x3ce
net/http.(*Client).Do(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:509
github.com/rclone/rclone/lib/rest.(*Client).Call(0xc0005a8c80, 0x1a28100, 0xc0005a7ec0, 0xc0005b4240, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/rest/rest.go:280 +0x91c
github.com/rclone/rclone/backend/googlephotos.(*Object).Update.func1(0x203000, 0xc0005b4360, 0x120)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:957 +0x6b
github.com/rclone/rclone/fs.pacerInvoker(0x1, 0x1, 0xc000f3e200, 0x8f00000001642a80, 0xc00077b790, 0x40cbd8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fs.go:1417 +0x3c
github.com/rclone/rclone/lib/pacer.(*Pacer).call(0xc00059d740, 0xc000f3e200, 0x1, 0x17, 0xc0005b43f8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:197 +0x8b
github.com/rclone/rclone/lib/pacer.(*Pacer).CallNoRetry(...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:225
github.com/rclone/rclone/backend/googlephotos.(*Object).Update(0xc000ac90a0, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc0011c4a90, 0x7f27628b38b8, 0xc000c58ae0, 0xc00038e4b0, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:956 +0x50b
github.com/rclone/rclone/backend/googlephotos.(*Fs).Put(0xc0005b7b90, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc0011c4a90, 0x7f27628b38b8, 0xc000c58ae0, 0xc00038e4b0, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:623 +0x218
github.com/rclone/rclone/fs/operations.Copy(0x1a28100, 0xc0005a7ec0, 0x1a3d120, 0xc0005b7b90, 0x0, 0x0, 0xc000c56570, 0x21, 0x1a3d620, 0xc000c58ae0, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:425 +0x152c
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairCopyOrMove(0xc0002b61c0, 0x1a28100, 0xc0005a7ec0, 0xc0005a8dc0, 0x1a3d120, 0xc0005b7b90, 0xc0002b6298)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:331 +0x21c
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startTransfers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:356 +0xb1

goroutine 88 [select]:
net/http.(*http2ClientConn).roundTrip(0xc001051e00, 0xc00009de00, 0x0, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:7573 +0x9a0
net/http.(*http2Transport).RoundTripOpt(0xc0005b9320, 0xc00009de00, 0xc000776a00, 0x725fc5, 0x0, 0xc00046a360)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6936 +0x172
net/http.(*http2Transport).RoundTrip(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6897
net/http.http2noDialH2RoundTripper.RoundTrip(0xc0005b9320, 0xc00009de00, 0xc000e85f80, 0x5, 0xc00046a3e8)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:9032 +0x3e
net/http.(*Transport).roundTrip(0xc00016c280, 0xc00009de00, 0x16c4b8f, 0xa, 0xc000d8e0d0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/transport.go:485 +0xdc1
net/http.(*Transport).RoundTrip(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/roundtrip.go:17
github.com/rclone/rclone/fs/fshttp.(*Transport).RoundTrip(0xc00034b380, 0xc00009de00, 0xc00009de00, 0x0, 0x150)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:350 +0x21a
github.com/rclone/rclone/vendor/golang.org/x/oauth2.(*Transport).RoundTrip(0xc00034b4a0, 0xc00009db00, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/golang.org/x/oauth2/transport.go:56 +0x124
net/http.send(0xc00009db00, 0x19f7cc0, 0xc00034b4a0, 0x0, 0x0, 0x0, 0xc000130098, 0x203000, 0x1, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:250 +0x443
net/http.(*Client).send(0xc00034b4d0, 0xc00009db00, 0x0, 0x0, 0x0, 0xc000130098, 0x0, 0x1, 0x1693780)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:174 +0xfa
net/http.(*Client).do(0xc00034b4d0, 0xc00009db00, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:641 +0x3ce
net/http.(*Client).Do(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:509
github.com/rclone/rclone/lib/rest.(*Client).Call(0xc0005a8c80, 0x1a28100, 0xc0005a7ec0, 0xc0005de360, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/rest/rest.go:280 +0x91c
github.com/rclone/rclone/backend/googlephotos.(*Object).Update.func1(0x203000, 0xc0005de480, 0x120)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:957 +0x6b
github.com/rclone/rclone/fs.pacerInvoker(0x1, 0x1, 0xc00005b640, 0xf900000001642a80, 0xc000777790, 0x40cbd8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fs.go:1417 +0x3c
github.com/rclone/rclone/lib/pacer.(*Pacer).call(0xc00059d740, 0xc00005b640, 0x1, 0x17, 0xc0005de518)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:197 +0x8b
github.com/rclone/rclone/lib/pacer.(*Pacer).CallNoRetry(...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:225
github.com/rclone/rclone/backend/googlephotos.(*Object).Update(0xc000e2cc40, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc000fb8b60, 0x7f27628b38b8, 0xc000c59260, 0xc0003ce040, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:956 +0x50b
github.com/rclone/rclone/backend/googlephotos.(*Fs).Put(0xc0005b7b90, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc000fb8b60, 0x7f27628b38b8, 0xc000c59260, 0xc0003ce040, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:623 +0x218
github.com/rclone/rclone/fs/operations.Copy(0x1a28100, 0xc0005a7ec0, 0x1a3d120, 0xc0005b7b90, 0x0, 0x0, 0xc000c56930, 0x21, 0x1a3d620, 0xc000c59260, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:425 +0x152c
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairCopyOrMove(0xc0002b61c0, 0x1a28100, 0xc0005a7ec0, 0xc0005a8dc0, 0x1a3d120, 0xc0005b7b90, 0xc0002b6298)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:331 +0x21c
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startTransfers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:356 +0xb1

goroutine 89 [semacquire]:
sync.runtime_SemacquireMutex(0xc001051f54, 0xc00064c500, 0x1)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/sema.go:71 +0x47
sync.(*Mutex).lockSlow(0xc001051f50)
	/opt/hostedtoolcache/go/1.13.7/x64/src/sync/mutex.go:138 +0xfc
sync.(*Mutex).Lock(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/sync/mutex.go:81
net/http.(*http2ClientConn).roundTrip(0xc001051e00, 0xc000118300, 0x0, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:7513 +0x1221
net/http.(*http2Transport).RoundTripOpt(0xc0005b9320, 0xc000118300, 0xc00064ca00, 0x725fc5, 0x0, 0xc00046a360)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6936 +0x172
net/http.(*http2Transport).RoundTrip(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6897
net/http.http2noDialH2RoundTripper.RoundTrip(0xc0005b9320, 0xc000118300, 0xc000394900, 0x5, 0xc00046a3e8)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:9032 +0x3e
net/http.(*Transport).roundTrip(0xc00016c280, 0xc000118300, 0x16c4b8f, 0xa, 0xc000be8e90)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/transport.go:485 +0xdc1
net/http.(*Transport).RoundTrip(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/roundtrip.go:17
github.com/rclone/rclone/fs/fshttp.(*Transport).RoundTrip(0xc00034b380, 0xc000118300, 0xc000118300, 0x0, 0x150)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:350 +0x21a
github.com/rclone/rclone/vendor/golang.org/x/oauth2.(*Transport).RoundTrip(0xc00034b4a0, 0xc000118200, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/vendor/golang.org/x/oauth2/transport.go:56 +0x124
net/http.send(0xc000118200, 0x19f7cc0, 0xc00034b4a0, 0x0, 0x0, 0x0, 0xc0003dd080, 0x203000, 0x1, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:250 +0x443
net/http.(*Client).send(0xc00034b4d0, 0xc000118200, 0x0, 0x0, 0x0, 0xc0003dd080, 0x0, 0x1, 0x1693780)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:174 +0xfa
net/http.(*Client).do(0xc00034b4d0, 0xc000118200, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:641 +0x3ce
net/http.(*Client).Do(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/client.go:509
github.com/rclone/rclone/lib/rest.(*Client).Call(0xc0005a8c80, 0x1a28100, 0xc0005a7ec0, 0xc000e00900, 0x0, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/rest/rest.go:280 +0x91c
github.com/rclone/rclone/backend/googlephotos.(*Object).Update.func1(0x203000, 0xc000e00a20, 0x120)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:957 +0x6b
github.com/rclone/rclone/fs.pacerInvoker(0x1, 0x1, 0xc0001acd00, 0xfd00000001642a80, 0xc00064d790, 0x40cbd8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fs.go:1417 +0x3c
github.com/rclone/rclone/lib/pacer.(*Pacer).call(0xc00059d740, 0xc0001acd00, 0x1, 0x17, 0xc000e00ab8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:197 +0x8b
github.com/rclone/rclone/lib/pacer.(*Pacer).CallNoRetry(...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:225
github.com/rclone/rclone/backend/googlephotos.(*Object).Update(0xc00119ad20, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc00085e9c0, 0x7f27628b38b8, 0xc000c7c900, 0xc000093530, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:956 +0x50b
github.com/rclone/rclone/backend/googlephotos.(*Fs).Put(0xc0005b7b90, 0x1a28100, 0xc0005a7ec0, 0x19f6580, 0xc00085e9c0, 0x7f27628b38b8, 0xc000c7c900, 0xc000093530, 0x1, 0x1, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:623 +0x218
github.com/rclone/rclone/fs/operations.Copy(0x1a28100, 0xc0005a7ec0, 0x1a3d120, 0xc0005b7b90, 0x0, 0x0, 0xc000c57470, 0x21, 0x1a3d620, 0xc000c7c900, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:425 +0x152c
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairCopyOrMove(0xc0002b61c0, 0x1a28100, 0xc0005a7ec0, 0xc0005a8dc0, 0x1a3d120, 0xc0005b7b90, 0xc0002b6298)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:331 +0x21c
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startTransfers
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:356 +0xb1

goroutine 91 [select, 49 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:147 +0x165
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 92 [select]:
github.com/rclone/rclone/fs/sync.(*pipe).Put(0xc0005a8dc0, 0x1a28100, 0xc0005a7ec0, 0x1a3d620, 0xc000ae9b60, 0x0, 0x0, 0xc000ae9b60)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:101 +0x1ec
github.com/rclone/rclone/fs/sync.(*syncCopyMove).SrcOnly(0xc0002b61c0, 0x7f27628b3840, 0xc000ae9b60, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:831 +0x314
github.com/rclone/rclone/fs/march.(*March).processJob(0xc0005b7c20, 0xc000395df5, 0x6, 0xc000395df5, 0x6, 0x7ffffffd, 0x0, 0x100, 0x0, 0x0, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:429 +0x345
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:154 +0x213
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 93 [select]:
github.com/rclone/rclone/fs/sync.(*pipe).Put(0xc0005a8dc0, 0x1a28100, 0xc0005a7ec0, 0x1a3d620, 0xc000848840, 0x0, 0x0, 0xc000848840)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:101 +0x1ec
github.com/rclone/rclone/fs/sync.(*syncCopyMove).SrcOnly(0xc0002b61c0, 0x7f27628b3840, 0xc000848840, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:831 +0x314
github.com/rclone/rclone/fs/march.(*March).processJob(0xc0005b7c20, 0xc000375a35, 0x6, 0xc000375a35, 0x6, 0x7ffffffd, 0x0, 0x100, 0x0, 0x0, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:429 +0x345
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:154 +0x213
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 94 [select, 87 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:147 +0x165
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 95 [select, 56 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:147 +0x165
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 96 [select, 87 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:147 +0x165
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 97 [select, 87 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:147 +0x165
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 98 [select]:
github.com/rclone/rclone/fs/sync.(*pipe).Put(0xc0005a8dc0, 0x1a28100, 0xc0005a7ec0, 0x1a3d620, 0xc0003d69c0, 0x0, 0x0, 0xc0003d69c0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/pipe.go:101 +0x1ec
github.com/rclone/rclone/fs/sync.(*syncCopyMove).SrcOnly(0xc0002b61c0, 0x7f27628b3840, 0xc0003d69c0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:831 +0x314
github.com/rclone/rclone/fs/march.(*March).processJob(0xc0005b7c20, 0xc000394e65, 0x6, 0xc000394e65, 0x6, 0x7ffffffd, 0x0, 0x100, 0x0, 0x0, ...)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:429 +0x345
github.com/rclone/rclone/fs/march.(*March).Run.func1(0xc000132b10, 0xc0005b7c20, 0xc00059d980, 0xc000132ad8, 0xc0001451a0, 0xc000132b00, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:154 +0x213
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:144 +0x1d1

goroutine 99 [chan receive, 87 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func2(0xc0005b7c20, 0xc00059d980, 0xc000132b20)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:196 +0x4e
created by github.com/rclone/rclone/fs/march.(*March).Run
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/march/march.go:194 +0x30a

goroutine 22026 [runnable]:
runtime.goparkunlock(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/proc.go:310
time.Sleep(0x987aff)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/time.go:105 +0x157
github.com/rclone/rclone/lib/pacer.(*Pacer).beginCall.func1(0xc00059d740, 0x987aff)
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:167 +0x2b
created by github.com/rclone/rclone/lib/pacer.(*Pacer).beginCall
	/home/runner/work/rclone/src/github.com/rclone/rclone/lib/pacer/pacer.go:166 +0x90

goroutine 21933 [select]:
github.com/rclone/rclone/fs/accounting.(*Account).averageLoop(0xc00085e8f0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:146 +0x140
created by github.com/rclone/rclone/fs/accounting.newAccountSizeName
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:66 +0x1e9

goroutine 22023 [chan receive]:
github.com/rclone/rclone/backend/local.(*fadvise).worker(0xc0001accc0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:114 +0x5a
created by github.com/rclone/rclone/backend/local.newFadvise
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:74 +0xff

goroutine 22025 [select]:
github.com/rclone/rclone/fs/asyncreader.(*AsyncReader).init.func1(0xc00119acb0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/asyncreader/asyncreader.go:83 +0x131
created by github.com/rclone/rclone/fs/asyncreader.(*AsyncReader).init
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/asyncreader/asyncreader.go:78 +0x18c

goroutine 22024 [select]:
github.com/rclone/rclone/fs/accounting.(*Account).averageLoop(0xc00085e9c0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:146 +0x140
created by github.com/rclone/rclone/fs/accounting.newAccountSizeName
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:66 +0x1e9

goroutine 1385 [select]:
github.com/rclone/rclone/fs/accounting.(*Account).averageLoop(0xc0006baf70)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:146 +0x140
created by github.com/rclone/rclone/fs/accounting.newAccountSizeName
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:66 +0x1e9

goroutine 21988 [chan receive]:
github.com/rclone/rclone/backend/local.(*fadvise).worker(0xc000f3e1c0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:114 +0x5a
created by github.com/rclone/rclone/backend/local.newFadvise
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:74 +0xff

goroutine 22005 [IO wait]:
internal/poll.runtime_pollWait(0x7f2762a62a98, 0x77, 0xffffffffffffffff)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc0010ea418, 0x77, 0x3400, 0x7434, 0xffffffffffffffff)
	/opt/hostedtoolcache/go/1.13.7/x64/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitWrite(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/internal/poll/fd_poll_runtime.go:96
internal/poll.(*FD).Write(0xc0010ea400, 0xc000726000, 0x4016, 0x8000, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/internal/poll/fd_unix.go:276 +0x2a9
net.(*netFD).Write(0xc0010ea400, 0xc000726000, 0x4016, 0x8000, 0x8, 0x8, 0xc000726005)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/fd_unix.go:220 +0x4f
net.(*conn).Write(0xc000130070, 0xc000726000, 0x4016, 0x8000, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/net.go:196 +0x68
github.com/rclone/rclone/fs/fshttp.(*timeoutConn).readOrWrite(0xc00039ab20, 0xc0009dd9c0, 0xc000726000, 0x4016, 0x8000, 0x1f, 0x0, 0x0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:75 +0x48
github.com/rclone/rclone/fs/fshttp.(*timeoutConn).Write(0xc00039ab20, 0xc000726000, 0x4016, 0x8000, 0xc0003d4dc0, 0xc0003d4dc0, 0x1753fe8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:92 +0x8a
crypto/tls.(*Conn).write(0xc0000a9880, 0xc000726000, 0x4016, 0x8000, 0xc00071c000, 0x4000, 0x5000)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:915 +0x160
crypto/tls.(*Conn).writeRecordLocked(0xc0000a9880, 0x17, 0xc00071c000, 0x4009, 0x5000, 0x3, 0x3, 0xc0009ddc72)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:964 +0x29f
crypto/tls.(*Conn).Write(0xc0000a9880, 0xc00071c000, 0x4009, 0x5000, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:1133 +0x1b7
net/http.http2stickyErrWriter.Write(0x19f5fa0, 0xc0000a9880, 0xc001051f58, 0xc00071c000, 0x4009, 0x5000, 0x0, 0x0, 0x1753fe8)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:6860 +0x68
bufio.(*Writer).Write(0xc0005a6dc0, 0xc00071c000, 0x4009, 0x5000, 0x1753fe8, 0xc0009ddd00, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/bufio/bufio.go:625 +0x144
net/http.(*http2Framer).endWrite(0xc000682460, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:1606 +0xab
net/http.(*http2Framer).WriteDataPadded(0xc000682460, 0xb39, 0xc00073c000, 0x4000, 0x4000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:1928 +0x25b
net/http.(*http2Framer).WriteData(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:1886
net/http.(*http2clientStream).writeRequestBody(0xc000e8cdc0, 0x7f2762aa79b8, 0xc0003ce0f0, 0x7f2762aa79d8, 0xc0003ce0f0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:7759 +0x337
net/http.(*http2Transport).getBodyWriterState.func1()
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:8951 +0xc5
created by net/http.http2bodyWriterState.scheduleBodyWrite
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:8998 +0xa1

goroutine 21989 [select]:
github.com/rclone/rclone/fs/accounting.(*Account).averageLoop(0xc0011c4a90)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:146 +0x140
created by github.com/rclone/rclone/fs/accounting.newAccountSizeName
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:66 +0x1e9

goroutine 21898 [chan receive]:
github.com/rclone/rclone/backend/local.(*fadvise).worker(0xc00005b5c0)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:114 +0x5a
created by github.com/rclone/rclone/backend/local.newFadvise
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:74 +0xff

goroutine 21899 [select]:
github.com/rclone/rclone/fs/accounting.(*Account).averageLoop(0xc000fb8b60)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:146 +0x140
created by github.com/rclone/rclone/fs/accounting.newAccountSizeName
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/accounting/accounting.go:66 +0x1e9

goroutine 15677 [IO wait]:
internal/poll.runtime_pollWait(0x7f2762a62a98, 0x72, 0xffffffffffffffff)
	/opt/hostedtoolcache/go/1.13.7/x64/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc0010ea418, 0x72, 0xf00, 0xfff, 0xffffffffffffffff)
	/opt/hostedtoolcache/go/1.13.7/x64/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0010ea400, 0xc0001f7000, 0xfff, 0xfff, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/internal/poll/fd_unix.go:169 +0x1cf
net.(*netFD).Read(0xc0010ea400, 0xc0001f7000, 0xfff, 0xfff, 0xc000bae480, 0x300000002, 0xc000bae480)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc000130070, 0xc0001f7000, 0xfff, 0xfff, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/net.go:184 +0x68
github.com/rclone/rclone/fs/fshttp.(*timeoutConn).readOrWrite(0xc00039ab20, 0xc000333888, 0xc0001f7000, 0xfff, 0xfff, 0x203000, 0xc0003338b8, 0xc0003338a8)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:75 +0x48
github.com/rclone/rclone/fs/fshttp.(*timeoutConn).Read(0xc00039ab20, 0xc0001f7000, 0xfff, 0xfff, 0x45d964b78a, 0x45d964b78a, 0x10)
	/home/runner/work/rclone/src/github.com/rclone/rclone/fs/fshttp/http.go:87 +0x8a
crypto/tls.(*atLeastReader).Read(0xc000638900, 0xc0001f7000, 0xfff, 0xfff, 0x538fe3, 0xc0010ea400, 0xc000333970)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:780 +0x60
bytes.(*Buffer).ReadFrom(0xc0000a9ad8, 0x19f5fc0, 0xc000638900, 0x40a345, 0x149a800, 0x15d1ea0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/bytes/buffer.go:204 +0xb4
crypto/tls.(*Conn).readFromUntil(0xc0000a9880, 0x7f2762aa7740, 0xc00039ab20, 0x5, 0xc00039ab20, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:802 +0xec
crypto/tls.(*Conn).readRecordOrCCS(0xc0000a9880, 0x0, 0x0, 0x1000)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:609 +0x124
crypto/tls.(*Conn).readRecord(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:577
crypto/tls.(*Conn).Read(0xc0000a9880, 0xc00068a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/crypto/tls/conn.go:1255 +0x161
bufio.(*Reader).Read(0xc000a9a180, 0xc000682498, 0x9, 0x9, 0xc000333d10, 0x0, 0x718192)
	/opt/hostedtoolcache/go/1.13.7/x64/src/bufio/bufio.go:226 +0x26a
io.ReadAtLeast(0x19f5dc0, 0xc000a9a180, 0xc000682498, 0x9, 0x9, 0x9, 0xc000092040, 0x0, 0x19f6200)
	/opt/hostedtoolcache/go/1.13.7/x64/src/io/io.go:310 +0x87
io.ReadFull(...)
	/opt/hostedtoolcache/go/1.13.7/x64/src/io/io.go:329
net/http.http2readFrameHeader(0xc000682498, 0x9, 0x9, 0x19f5dc0, 0xc000a9a180, 0x0, 0x0, 0xb6, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:1477 +0x87
net/http.(*http2Framer).ReadFrame(0xc000682460, 0xc0008093e0, 0x0, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:1735 +0xa1
net/http.(*http2clientConnReadLoop).run(0xc000333fb8, 0x1000000010000, 0x0)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:8175 +0x8e
net/http.(*http2ClientConn).readLoop(0xc001051e00)
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:8103 +0xa3
created by net/http.(*http2Transport).newClientConn
	/opt/hostedtoolcache/go/1.13.7/x64/src/net/http/h2_bundle.go:7162 +0x62f

goroutine 21932 [chan receive]:
github.com/rclone/rclone/backend/local.(*fadvise).worker(0xc0001ac180)
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:114 +0x5a
created by github.com/rclone/rclone/backend/local.newFadvise
	/home/runner/work/rclone/src/github.com/rclone/rclone/backend/local/fadvise_unix.go:74 +0xff

$  rclone version
rclone v1.51.0
- os/arch: linux/amd64
- go version: go1.13.7
```

**Comment by ncw** [comment]
@jpkrohling can you stick that in a new issue please as I think it is unrelated to this one then I'll have a go at fixing - thanks :-)

**Comment by jpkrohling** [comment]
Done, created #4003

**Comment by eclipseo** [comment]
> @eclipseo, @ncw, how hard would it be to get this fix packaged as an update for Fedora 31? Looks like Fedora 31 is at 1.50 ([`rclone-1.50.2-1.fc31`](https://koji.fedoraproject.org/koji/buildinfo?buildID=1428911)), and this fix should be available as part of 1.51.

@jpkrohling  I was away for awhile for health reasons, I intend to update to the last release version as soon as I have the new deps in.

**Comment by jpkrohling** [comment]
I hope everything is fine now! I see that 1.51 is now available in Fedora 32. Thank you for that, it's much appreciated!
